src/libproc_macro/lib.rs

```diff
@@ -125,6 +125,10 @@ pub mod __internal {
         fn register_attr_proc_macro(&mut self,
                                     name: &str,
                                     expand: fn(TokenStream, TokenStream) -> TokenStream);
+
+        fn register_bang_proc_macro(&mut self,
```

Comment:
For `#[proc_macro_derive(A)] fn f(..) { .. }`, we would say `A` is a *[derive] proc macro* and `f` is the *[derive] proc macro function*.